elastic/kibana
# [monitoring] Rework Watcher migration flow

# Summary

The Stack Monitoring UI had a code path where, when the default alerts were being created it was meant to disable the Elasticsearch Watches that are installed when Internal Collection is enabled.
However, a recent support escalation showed that Kibana was calling the wrong Elasticsearch API. Further investigation made the situation worse by highlighting that the API in question had to be invoked on the _**monitored**_ cluster(s), and not the monitoring cluster.

To address this short coming, this PR pulls out the code that tried to disable the Watches and instead introduces a UI flow that guides the user to perform this migration before setting up the Kibana rules.

The flow now includes a new modal with instructions for why and how to migrate. This modal is shown after the user confirms to create the default rules in the old modal or when clicking "Create default rules" in the top nav.

Reworded landing screen modal:
<img width="982" alt="Screenshot 2023-06-15 at 21 58 30" src="https://github.com/elastic/kibana/assets/2564140/9ffa3fdc-2c29-4113-938a-cd590e414cfd">

New modal can also be triggered from top nav:
<img width="427" alt="Screenshot 2023-06-15 at 21 59 02" src="https://github.com/elastic/kibana/assets/2564140/9bfc8a71-e387-4682-9818-a1fa7474e038">

New modal with migration instructions:
<img width="977" alt="Screenshot 2023-06-15 at 21 58 41" src="https://github.com/elastic/kibana/assets/2564140/a8437b23-9bb9-43e7-8ee7-51035c367b0c">

# How to test
Start a local Elasticsearch instance, connect Kibana and collect some monitoring data (Internal Collection is quick and easy).
Once data is available the modal should open automatically when visiting the Stack Monitoring app.
It should be possible to close the modal and have it open again in the next session (tracked in session storage).
If the user confirms both modals there should be a toast confirming the creation of the rules and the rules should appear under `Stack Management > Rules`
